Lib/pdb.py

```diff
@@ -588,6 +588,36 @@ def _hold_exceptions(self, exceptions):
             self._chained_exceptions = tuple()
             self._chained_exception_index = 0
 
+    def process_cmdqueue(self):
+        """Process commands in the command queue.
+
+        The function returns in two cases:
+            1. the command queue is empty
+            2. a command to continue execution is encountered
+
+        The return value is
+            True - if the user input is expected after processing,
+            False - otherwise.
+        """
+        if not self.cmdqueue:
+            return True
+
+        # It's possible that cmdqueue does not contain any command that resumes
+        # execution, and _cmdloop will stuck in the loop waiting for user input
+        # in that case. To prevent that, we append a sentinel command (step) at
+        # the end of the queue. This makes sure we can break out of the loop.
+        # If the sentinel command is consumed, we know extra user input is
+        # needed. Otherwise we can pop the sentinel command and return.
```

Comment:
```suggestion
        # the end of the queue. If the queue is empty after it is processed, it
        # means that the sentinel command was consumed, i.e., the last
        # command in the queue resumed execution. Otherwise, we pop
        # the sentinel and return.
```